app/src/main/java/edu/usf/sas/pal/muser/firebase/model/Event.kt

```diff
@@ -0,0 +1,22 @@
+package edu.usf.sas.pal.muser.firebase.model
+
+/**
+ * A model class that stores the actions performed on the song track. The actions like play, pause,
+ * skip, repeat and seek are stored along with the nested [SongData] class.
+ */
+
+data class Event
+
+/**
+ * [event] - The event attribute includes events like play, pause, skip, repeat and seek.
+ * [currentTimeMs] - Stores the timestamp of the action recorded.
+ * [nanoTime] - Stores the  value of the running JVM's time source in nanoseconds.
+ * [song] - Song on which the action was performed.
+ */
+(val event:String,
+ val currentTimeMs: Long,
+ val nanoTime: Long,
+ val startTime: Long?,
```

Comment:
Oops - that should be `elapsedTime`. Also, I kept them nullable since the values are dependent upon the `song` object and the values would be null if the `song` object was null. Should I keep them mandatory too?

Reply:
Yes, if we're always going to be capturing the data, then it shouldn't be nullable (i.e., you should remove the `?`).